store-implementation/parquet-store/src/main/java/uk/gov/gchq/gaffer/parquetstore/ParquetStore.java

```diff
@@ -0,0 +1,335 @@
+import org.apache.hadoop.conf.Configuration;
+import org.apache.hadoop.fs.FSDataInputStream;
+import org.apache.hadoop.fs.FileStatus;
+import org.apache.hadoop.fs.FileSystem;
+import org.apache.hadoop.fs.Path;
+import org.apache.parquet.bytes.BytesUtils;
+import org.apache.parquet.io.api.Binary;
+import org.apache.spark.SparkConf;
+import org.apache.spark.sql.SparkSession;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import scala.Tuple3;
+import uk.gov.gchq.gaffer.commonutil.iterable.CloseableIterable;
+import uk.gov.gchq.gaffer.data.element.Element;
+import uk.gov.gchq.gaffer.operation.Operation;
+import uk.gov.gchq.gaffer.operation.impl.add.AddElements;
+import uk.gov.gchq.gaffer.operation.impl.get.GetElements;
+import uk.gov.gchq.gaffer.parquetstore.operation.addelements.handler.AddElementsFromRDDHandler;
+import uk.gov.gchq.gaffer.parquetstore.operation.addelements.handler.AddElementsHandler;
+import uk.gov.gchq.gaffer.parquetstore.operation.getelements.handler.DummyGetAdjacentEntitySeedsHandler;
+import uk.gov.gchq.gaffer.parquetstore.operation.getelements.handler.GetAllElementsHandler;
+import uk.gov.gchq.gaffer.parquetstore.operation.getelements.handler.GetDataframeOfElementsHandler;
+import uk.gov.gchq.gaffer.parquetstore.operation.getelements.handler.GetElementsHandler;
+import uk.gov.gchq.gaffer.parquetstore.serialisation.ParquetSerialisationFactory;
+import uk.gov.gchq.gaffer.parquetstore.utils.Constants;
+import uk.gov.gchq.gaffer.parquetstore.utils.SchemaUtils;
+import uk.gov.gchq.gaffer.serialisation.Serialiser;
+import uk.gov.gchq.gaffer.spark.SparkUser;
+import uk.gov.gchq.gaffer.spark.operation.dataframe.GetDataFrameOfElements;
+import uk.gov.gchq.gaffer.spark.operation.scalardd.ImportRDDOfElements;
+import uk.gov.gchq.gaffer.store.Context;
+import uk.gov.gchq.gaffer.store.Store;
+import uk.gov.gchq.gaffer.store.StoreException;
+import uk.gov.gchq.gaffer.store.StoreProperties;
+import uk.gov.gchq.gaffer.store.StoreTrait;
+import uk.gov.gchq.gaffer.store.operation.handler.OperationHandler;
+import uk.gov.gchq.gaffer.store.operation.handler.OutputOperationHandler;
+import uk.gov.gchq.gaffer.store.schema.Schema;
+import uk.gov.gchq.gaffer.store.schema.SchemaOptimiser;
+import uk.gov.gchq.gaffer.user.User;
+
+import java.io.IOException;
+import java.nio.charset.Charset;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Comparator;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Set;
+
+/**
+ *
+ */
+public class ParquetStore extends Store {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(ParquetStore.class);
+    private static final Set<StoreTrait> TRAITS = new HashSet<>();
+    private HashMap<String, ArrayList<Tuple3<Object[], Object[], String>>> groupToIndex;
+    private SchemaUtils schemaUtils;
+    private FileSystem fs;
+    private long currentSnapshot;
+
+    static {
+        TRAITS.add(StoreTrait.INGEST_AGGREGATION);
```

Comment:
Please could you define your traits like we do in the accumulo store:

 public static final Set<StoreTrait> TRAITS =
            Collections.unmodifiableSet(Sets.newHashSet(
                    ORDERED,
                    VISIBILITY,
                    INGEST_AGGREGATION,
                    QUERY_AGGREGATION,
                    PRE_AGGREGATION_FILTERING,
                    POST_AGGREGATION_FILTERING,
                    POST_TRANSFORMATION_FILTERING,
                    TRANSFORMATION,
                    STORE_VALIDATION
            ));

i.e wrap them in an unmodifiable set.